src/Psalm/Internal/Provider/ReturnTypeProvider/ReflectionParameterGetType.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psalm\Internal\Provider\ReturnTypeProvider;
+
+use Psalm\Plugin\EventHandler\Event\MethodReturnTypeProviderEvent;
+use Psalm\Type;
+use ReflectionType;
+use function array_key_exists;
+
+class ReflectionParameterGetType implements \Psalm\Plugin\EventHandler\MethodReturnTypeProviderInterface
+{
+    public static function getClassLikeNames() : array
+    {
+        return ['ReflectionParameter'];
+    }
+
+    public static function getMethodReturnType(MethodReturnTypeProviderEvent $event) : ?Type\Union
+    {
+        if ($event->getMethodNameLowercase() !== 'gettype') {
+            return null;
+        }
+
+        $stmt = $event->getStmt();
+        if (!$stmt instanceof \PhpParser\Node\Expr\MethodCall) {
+            return null;
+        }
+
+        if (!$stmt->var instanceof \PhpParser\Node\Expr\Variable) {
+            return null;
+        }
+
+        if (!is_string($stmt->var->name)) {
+            return null;
+        }
+
+        $scopedVarName = '$' . $stmt->var->name . '->hastype()';
+        if (!isset($event->getContext()->vars_in_scope[$scopedVarName])) {
+            return null;
+        }
+
+        $type = $event->getContext()->vars_in_scope[$scopedVarName];
+
+        if ($type->isTrue()) {
+            return Type::parseString(ReflectionType::class);
```

Comment:
```suggestion
            return new Type\Union([
                new Type\Atomic\TNamedObject(ReflectionType::class)
            ]);
```
There's no need to go through parser as we know exactly what we need to return.

Reply:
Ah yes, I tried initially, but couldn't find out how to do it. Thanks!